pkg/frontend/v2/frontend.go

```diff
@@ -200,6 +218,22 @@ func (f *Frontend) RoundTripGRPC(ctx context.Context, req *httpgrpc.HTTPRequest)
 		}
 	}
 
+	f.mtx.Lock()
+	if f.stopped {
+		f.mtx.Unlock()
+		return nil, fmt.Errorf("frontend not running")
+	}
+
+	f.inflightRequests++
+	f.mtx.Unlock()
+	defer func() {
+		f.mtx.Lock()
+		f.inflightRequests--
+		// Wake up the stopped method if it's waiting
```

Comment:
[nit] No need to describe implementation. (Also the method is named `stopping`).

Reply:
Removed.